src/main/java/org/jabref/gui/exporter/CreateModifyExporterDialogViewModel.java

```diff
@@ -0,0 +1,115 @@
+ *
+ */
+
+public class CreateModifyExporterDialogViewModel extends AbstractViewModel {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateModifyExporterDialogViewModel.class);
+
+    private final DialogService dialogService;
+    private final PreferencesService preferences;
+
+    private final StringProperty name = new SimpleStringProperty("");
+    private final StringProperty layoutFile = new SimpleStringProperty("");
+    private final StringProperty extension = new SimpleStringProperty("");
+
+    private final JournalAbbreviationLoader loader;
+
+
+    public CreateModifyExporterDialogViewModel(Optional<ExporterViewModel> exporter, DialogService dialogService, PreferencesService preferences,
+                                               JournalAbbreviationLoader loader) {
+        this.dialogService = dialogService;
+        this.preferences = preferences;
+        this.loader = loader;
+
+        setTextFields(exporter);
+    }
+
+    public Optional<ExporterViewModel> saveExporter() {
+        Path layoutFileDir = Paths.get(layoutFile.get()).getParent();
+        if (layoutFileDir != null) {
+            String layoutFileDirString = layoutFileDir.toString();
+            preferences.setExportWorkingDirectory(layoutFileDirString);
+        }
+
+        // Check that there are no empty strings.
+        if (layoutFile.get().isEmpty() || name.get().isEmpty() || extension.get().isEmpty()
+            || !layoutFile.get().endsWith(".layout")) {
+
+            LOGGER.info("One of the fields is empty or invalid!");
+            // Return empty exporter to the main exporter customization dialog
+            return Optional.empty();
+        }
+
+        // Create a new exporter to be returned to ExportCustomizationDialogViewModel, which requested it
+        LayoutFormatterPreferences layoutPreferences = preferences.getLayoutFormatterPreferences(loader);
+        SavePreferences savePreferences = preferences.loadForExportFromPreferences();
+        TemplateExporter format = new TemplateExporter(name.get(), layoutFile.get(), extension.get(),
+                                                       layoutPreferences, savePreferences);
+        format.setCustomExport(true);
+        return Optional.of(new ExporterViewModel(format));
+    }
+
+    public String getExportWorkingDirectory() { //i.e. layout dir
+        return preferences.getExportWorkingDirectory();
+    }
+
+    public void browse() {
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+            .addExtensionFilter(Localization.lang("Custom layout file"), StandardFileType.LAYOUT)
+            .withDefaultExtension(Localization.lang("Custom layout file"), StandardFileType.LAYOUT)
+            .withInitialDirectory(getExportWorkingDirectory()).build();
+        dialogService.showFileOpenDialog(fileDialogConfiguration).ifPresent(f -> layoutFile.set(f.toAbsolutePath().toString())); //implement setting the text
```

Comment:
I put it in both, since saving the exporter might also indicate where the working directory should be. However, now it won't set the directory as before. Since this fix isn't working I am pushing this into a separate branch `exporter_to_javafx_setwd` (in my fork, but not in this pull request). I will try to get to the other changes you and Christoph mentioned later.